src/__tests__/waitFor.test.tsx

```diff
@@ -78,6 +78,38 @@ test('waits for element with custom interval', async () => {
   expect(mockFn).toHaveBeenCalledTimes(2);
 });
 
+test('waits for async event with fireEvent', async () => {
+  const Comp = ({ onPress }: { onPress: () => void }) => {
```

Comment:
readability: Can we move the `Comp` outside of `test` block?

Reply:
Sure, I prefer colocation and I find it easier to read (since you can fold the `test()` block), but since we're doing that, I guess consistency is also worth!